x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/FunctionInfo.java

```diff
@@ -23,4 +23,6 @@
     String description() default "";
 
     boolean isAggregation() default false;
+
+    String minArgs() default "";
```

Comment:
The type and what it represents is confusing. `minArgs` indicates a number while the type is String?
And looking at `show.csv-spec` I see mostly `null` values and here and there `1` for `minArgs`. What does `null` represent?

Reply:
Here's the `minArgs` values defined:
* `concat`: 2
* `case`: 3
* `coalesce`: 1
* `greatest`: 1
* `least`: 1
* `cidr_match`: 2

I had to adopt `null` here to make tests pass. Empty values were my initial implementation but tests were complaining. With `null` it passes.

Reply:
> The type and what it represents is confusing. minArgs indicates a number while the type is String?

I thought about adopting `int` at first, but given its small footprint using `String` avoided changes for most part of the code around. I can refactor to it, hoping all the testing logic won't be too affected.